handler/turtle.py

```diff
@@ -0,0 +1,274 @@
+
+def load(repo, batch_config):
+    return Batch(repo, batch_config)
+
+
+#============================================================================
+# BATCH CLASS (FOR PAGED BINARIES PLUS RDF METADATA)
+#============================================================================
+
+class Batch():
+
+    '''Iterator class representing a set of resources to be loaded'''
+
+    def __init__(self, repo, config):
+        self.logger = logging.getLogger(
+            __name__ + '.' + self.__class__.__name__
+            )
+
+        # Check for required configuration items and set up paths
+        required_keys = ['HANDLER',
+                         'COLLECTION',
+                         'ROOT',
+                         'MAPFILE',
+                         'BATCH_INDEX',
+                         'LOG_LOCATION',
+                         'LOG_CONFIG',
+                         'METADATA_FILE',
+                         'METADATA_PATH',
+                         'DATA_PATH'
+                         ] 
+        for key in required_keys:
+            if not config.get(key):
+                raise ConfigException(
+                    'Missing required key {0} in batch config'.format(key)
+                    )
+
+        # Set configuration Properties
+        self.local_path    = os.path.normpath(config.get('ROOT'))
+        self.index_file    = os.path.join(self.local_path,
+                                          config.get('BATCH_INDEX'))
+        self.data_path     = os.path.join(self.local_path, 
+                                          config['DATA_PATH'])
+        self.metadata_path = os.path.join(self.local_path, 
+                                          config['METADATA_PATH'])
+        self.metadata_file = os.path.join(self.local_path,
+                                          config.get('METADATA_FILE'))
+        self.collection    = pcdm.Collection.from_repository(repo,
+                                                     config.get('COLLECTION'))
+
+        # Create data structures to accumulate process results
+        self.items       = {}
+        self.incomplete  = []
+        self.extra_files = []
+
+        # Check for required metadata file and path
+        if not os.path.isdir(self.metadata_path):
+            os.mkdir(self.metadata_path)
+        if not os.path.isfile(self.metadata_file):
+            raise ConfigException('Specified metadata file could not be found')
+
+        # Generate index of all files in the data path
+        self.logger.info('Walking the "data path" tree to create a file index')
+        self.all_files = {}
+        for root, dirs, files in os.walk(self.data_path):
+            for f in files:
+                self.all_files[f] = os.path.join(root,f)
+        self.logger.info("Found {0} files".format(len(self.all_files)))
+
+        # Generate item-level metadata graphs and store as files
+        with open(self.metadata_file, 'r') as f:
```

Comment:
This whole operation of parsing the metadata graph and creating the individual turtle file seems like the kind of thing that should be able to be controlled separately from the loading of each individual item, so that it only has to be run once on the batch, and not for every time the loader is run.